docs/configurations.md

```diff
@@ -0,0 +1,77 @@
+BALENA_SUPERVISOR_DELTA_RETRY_COUNT | integer | 30 | Number of retry attempts for downloading deltas. | v6.2.0
+BALENA_SUPERVISOR_DELTA_RETRY_INTERVAL | integer | 10000 | Millisecond time before another delta download is attempted after failiure. | v6.2.0
+BALENA_SUPERVISOR_DELTA_VERSION | integer | 3 | Specifies the delta version to use which has different APIs and download algorithms. | v7.9.0
+BALENA_SUPERVISOR_PERSISTENT_LOGGING | boolean | false | Persists journal logs to disk so they will be available after a reboot event. | v7.15.0
+BALENA_SUPERVISOR_INSTANT_UPDATE_TRIGGER | boolean | true | Enable / Disable instant triggering of updates when a new release is deployed. If set to false, the device will ignore the notification that is triggered when the device's target state has changed. In this case, the device will rely on polling to apply updates. Coupled with a large BALENA_SUPERVISOR_POLL_INTERVAL, this allows spreading out updates in large fleets to avoid overloading local networks when there is a large number of devices at one location. | v9.13.0
+BALENA_HOST_FIREWALL_MODE | string | off | Toggle firewall modes between on, off, and auto. | v11.9.1
+BALENA_HOST_DISCOVERABILITY | boolean | true | Enable / Disable Avahi to run which will allow the device to respond to requests such as network scans. | v11.9.2
+BALENA_HOST_SPLASH_IMAGE | integer | /boot/splash/balena-logo-default.png | Allows changing splash screen on boot to user defined file from userspace. | v12.3.0
+---
+
+In addition to these values, there may be some device-type specific configuration variables that can be set.
+
+## Config.txt
+
+> Supported boards: Any Raspberry Pi and BalenaFin.
+
+This device type utilizes a `config.txt` [file](https://www.raspberrypi.org/documentation/configuration/config-txt/README.md) which we offer some values to be configured.
+
+Name | Type | Default | Description | Valid from
+--- | --- | --- | --- | ---
+BALENA_HOST_CONFIG_disable_splash | integer | 1 | Enable / Disable the Raspberry Pi rainbow splash screen. | v1.0.0
+BALENA_HOST_CONFIG_dtparam | string | "i2c_arm=on","spi=on","audio=on" | Define DT parameters. | v1.0.0
+BALENA_HOST_CONFIG_gpu_mem | integer | 16 | Define device GPU memory in megabytes. | v1.0.0
+
+You can find more information on updating `config.txt` through configuration variables in our [Advanced Boot Configuration Guide][boot-config-guide].
+
+## ConfigFS
+
+> Supported boards: UP Board
+
+This device type utilizes a `configFS` [file](https://www.raspberrypi.org/documentation/configuration/config-txt/README.md) interface that allows definition of arbitrary functions and configurations.
+
+Name | Type | Default | Description | Valid from
+--- | --- | --- | --- | ---
+BALENA_HOST_CONFIGFS_ssdt | string | undefined | Allows configuring a user defined SSDTs from userspace via the configfs interface. See [SSDT Overlays](https://www.kernel.org/doc/html/latest/admin-guide/acpi/ssdt-overlays.html#ssdt-overlays). | v10.9.2
```

Comment:
With any configs that support either strings (`"value1"`) or comma-separated strings (`"value1","value2"`), it would be clearer to have a distinction if communicating data types. In the code I see that config vars that support this format are 
- ConfigFS: 
   - *_HOST_CONFIGFS_ssdt
- Config.txt:
   - *_HOST_CONFIG_dtparam
   - *_HOST_CONFIG_dtoverlay
   - *_HOST_CONFIG_device_tree_param
   - *_HOST_CONFIG_device_tree_overlay
   - *_HOST_CONFIG_gpio
- extlinux:
   - *_HOST_EXTLINUX_isolcpus

as opposed to config vars like *_HOST_EXTLINUX_fdt which only support `"value1"` and not `"value1","value2"`. Currently both these data types are listed as "string" in these tables. What do you think?

Reply:
I see what you're saying but think strings is fine because "value1","value2" is still a string.